frontend/src/metabase/redux/embed/embed.ts

```diff
@@ -37,39 +48,48 @@ export const urlParameterToBoolean = (
   }
 };
 
+interface Location {
+  search: string;
+}
 const interactiveEmbedSlice = createSlice({
   name: "interactiveEmbed",
   initialState: {
     options: {} as InteractiveEmbeddingOptions,
     isEmbeddingSdk: false,
   },
-  reducers: {
-    setInitialUrlOptions: (
-      state,
-      action: PayloadAction<{ search: string }>,
-    ) => {
-      const searchOptions = compose(
-        normalizeEntityTypes,
-        excludeNonInteractiveEmbeddingOptions,
-        parseSearchOptions,
-        normalizeEntityTypesCommaSeparatedSearchParameter,
-      )(action.payload.search);
-
-      state.options = {
-        ...DEFAULT_INTERACTIVE_EMBEDDING_OPTIONS,
-        ...searchOptions,
-      };
-    },
-    setOptions: (
-      state,
-      action: PayloadAction<Partial<InteractiveEmbeddingOptions>>,
-    ) => {
-      state.options = {
-        ...state.options,
-        ...action.payload,
-      };
-    },
-  },
+  reducers: (create) => ({
+    setInitialUrlOptions: create.asyncThunk(
+      ({ search }: Location, { dispatch }) => {
+        const { entity_types, ...searchOptions } = compose(
+          normalizeEntityTypes,
+          excludeNonInteractiveEmbeddingOptions,
+          parseSearchOptions,
+          normalizeEntityTypesCommaSeparatedSearchParameter,
+        )(search);
+
+        dispatch(setEntityTypes(entity_types));
```

Comment:
I convert this action to async thunk so I could dispatch this action. The rest of the logic should be the same.